doc/manual/src/glossary.md

```diff
@@ -85,12 +85,18 @@
 
     [store path]: #gloss-store-path
 
+  - [file system object]{#gloss-store-object}\
+    The Nix data model for representing file system data.
+
+    See [File System Object](@docroot@/architecture/file-system-object.md) for details.
+
+    [file system object]: #gloss-file-system-object
+
   - [store object]{#gloss-store-object}\
-    A file that is an immediate child of the Nix store directory. These
-    can be regular files, but also entire directory trees. Store objects
-    can be sources (objects copied from outside of the store),
-    derivation outputs (objects produced by running a build task), or
-    derivations (files describing a build task).
+
+    A store object is a [file system object] with [reference]s to other store objects.
+
+    Store objects can be sources (copied from outside of the store), derivation outputs (produced by running a build task), or [derivation]s (files describing a build task).
```

Comment:
I think of "source" only as a relation between a derivation and a content addressed path that is available at evaluation time. I don't think it's a viable concept on its own.
What about `nix store add file` or whatever it's called?

```suggestion
    Store objects can contain content addressed data (such as sources added through evaluation), derivation outputs (produced by running a build task), or [derivation]s (files describing a build task).
```

Reply:
Agreed "sources" isn't a type of data, but a relationship between data and derivation. (`inputSrcs` field.)

I wouldn't mind just dropping this sentence until we make the next page for store objects.

I also think it's better to say "derivations can be persisted as store object" (i.e. as part of the derivation documentation) than "store objects can represent derivations". Store objects can represent *anything*, and the drv file serialization is just a mere *use* of that property.

Reply:
Yeah, let's drop the sentence.